javascript/encoders_test.js

```diff
@@ -0,0 +1,233 @@
+    const { configPath, modalityPaths, tokenizerPath } = await getModel(
+        modelId,
+        modalities,
+        hf_token,
+        '.onnx'
+    );
+
+    const imageProcessor = new ImageProcessor(configPath);
+    await imageProcessor.init();
+    const imageEncoder = new ImageEncoder(modalityPaths.image_encoder, imageProcessor);
+    await imageEncoder.init();
+    const textProcessor = new TextProcessor(configPath, tokenizerPath);
+    await textProcessor.init();
+    const textEncoder = new TextEncoder(modalityPaths.text_encoder, textProcessor);
+    await textEncoder.init();
+
+    const texts = [
+        "A group of friends enjoy a barbecue on a sandy beach, with one person grilling over a large black grill, while the other sits nearby, laughing and enjoying the camaraderie.",
+        "A white and orange cat stands on its hind legs, reaching towards a wicker basket filled with red raspberries on a wooden table in a garden, surrounded by orange flowers and a white teapot, creating a serene and whimsical scene.",
+        "A little girl in a yellow dress stands in a grassy field, holding an umbrella and looking at the camera, amidst rain.",
+        "This serene bedroom features a white bed with a black canopy, a gray armchair, a black dresser with a mirror, a vase with a plant, a window with white curtains, a rug, and a wooden floor, creating a tranquil and elegant atmosphere.",
+        "The image captures the iconic Louvre Museum in Paris, illuminated by warm lights against a dark sky, with the iconic glass pyramid in the center, surrounded by ornate buildings and a large courtyard, showcasing the museum's grandeur and historical significance.",
+    ];
+    const imageUrls = [
+        "https://github.com/ashvardanian/ashvardanian/blob/master/demos/bbq-on-beach.jpg?raw=true",
+        "https://github.com/ashvardanian/ashvardanian/blob/master/demos/cat-in-garden.jpg?raw=true",
+        "https://github.com/ashvardanian/ashvardanian/blob/master/demos/girl-and-rain.jpg?raw=true",
+        "https://github.com/ashvardanian/ashvardanian/blob/master/demos/light-bedroom-furniture.jpg?raw=true",
+        "https://github.com/ashvardanian/ashvardanian/blob/master/demos/louvre-at-night.jpg?raw=true",
+    ];
+
+    const textEmbeddings = [];
+    const imageEmbeddings = [];
+
+    for (let i = 0; i < texts.length; i++) {
+        const text = texts[i];
+        const imageUrl = imageUrls[i];
+        const imageBuffer = await fetchImage(imageUrl);
+
+        const processedText = await textProcessor.process(text);
+        const processedImage = await imageProcessor.process(imageBuffer);
+
+        const textEmbedding = await textEncoder.encode(processedText);
+        const imageEmbedding = await imageEncoder.encode(processedImage);
+
+        textEmbeddings.push(new Float32Array(textEmbedding.embeddings.cpuData));
+        imageEmbeddings.push(new Float32Array(imageEmbedding.embeddings.cpuData));
```

Comment:
```suggestion
        textEmbeddings.push(new Float32Array(textEmbedding.embeddings.data));
        imageEmbeddings.push(new Float32Array(imageEmbedding.embeddings.data));
```
You should be able to use `.data` (I think). Might safeguard in future if you support the dml or cuda backends.